internal/provider/waypoint/resource_waypoint_action.go

```diff
@@ -0,0 +1,629 @@
+	switch method {
+	case waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodGET:
+		methodString = types.StringValue("GET")
+	case waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPOST:
+		methodString = types.StringValue("POST")
+	case waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPUT:
+		methodString = types.StringValue("PUT")
+	case waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPATCH:
+		methodString = types.StringValue("PATCH")
+	case waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodDELETE:
+		methodString = types.StringValue("DELETE")
+	default:
+		return methodString, fmt.Errorf("unknown method")
+	}
+	return methodString, nil
+}
+
+func convertMethodToEnumType(method string) (waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethod, error) {
+	var methodEnum waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethod
+	switch method {
+	case "GET":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodGET
+	case "POST":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPOST
+	case "PUT":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPUT
+	case "PATCH":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodPATCH
+	case "DELETE":
+		methodEnum = waypoint_models.HashicorpCloudWaypointActionConfigFlavorCustomMethodDELETE
+	default:
+		return methodEnum, fmt.Errorf("unknown method")
+	}
+	return methodEnum, nil
+}
+
+func (r *ActionResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_waypoint_action"
+}
+
+func (r *ActionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "The Waypoint Action resource managed the lifecycle of an Action.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the Action.",
+				Computed:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Action.",
+				Required:    true,
+			},
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the HCP project where the Action is located.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"organization_id": schema.StringAttribute{
+				Description: "The ID of the HCP organization where the Action is located.",
+				Computed:    true,
+			},
+			// An Action description must be fewer than 125 characters if set.
+			"description": schema.StringAttribute{
+				Description: "A description of the Action.",
+				Optional:    true,
+			},
+			"namespace_id": schema.StringAttribute{
+				Description: "Internal Namespace ID",
```

Comment:
This could use a better description...I'm also not entirely sure where a user finds this value in HCP.

Reply:
It also looks like we don't use the namespace_id field anyway for our namespace, but use Org and Project IDs to look it up. So maybe this field goes away?

Reply:
It was only used to expose the namespace_id to the user, I never intended it for our use. Your comment made me realize that the field is not that useful however so I will just remove it.